monkey/infection_monkey/island_api_client/http_client.py

```diff
@@ -122,17 +125,15 @@ def _send_request(
         *args,
         **kwargs,
     ) -> requests.Response:
-        if self._api_url is None:
-            raise RuntimeError(
-                "HTTP client is not connected to the Island server,"
-                "establish a connection with 'connect()' before "
-                "attempting to send any requests"
-            )
-        url = f"{self._api_url}/{endpoint}".strip("/")
```

Comment:
Maybe this change could be reverted and query parameters should be passed as `params` keyword arguments. This will reduce changes to the UT's even further 🤔